pmlc/tools/pmlc-jit/pmlc-jit.cc

```diff
@@ -20,11 +27,47 @@
 #include "pmlc/util/all_dialects.h"
 #include "pmlc/util/all_passes.h"
 
+#define EXPECTED_EXCEPTION_FLG_START_LEN 35
+#define EXPECTED_EXCEPTION_FLAG_START "-expected-runtime-exception-string="
+
 int main(int argc, char **argv) {
   registerAllDialects();
   llvm::InitLLVM y(argc, argv);
   llvm::InitializeNativeTarget();
   llvm::InitializeNativeTargetAsmPrinter();
   mlir::initializeLLVMPasses();
-  return mlir::JitRunnerMain(argc, argv, nullptr);
+
+  // Get the expected-runtime-exception-string and remove the parameters
+  // unknown to the tool that is being delegated to.
+  std::string expectedString;
+  int fixedUpCounter = 0;
+
+  for (int i = 0; i < argc; i++) {
+    if (fixedUpCounter != i) {
+      argv[fixedUpCounter] = argv[i];
+    }
+
+    std::string arg = argv[i];
+    if (arg.find(EXPECTED_EXCEPTION_FLAG_START) == 0) {
+      expectedString = arg.substr(EXPECTED_EXCEPTION_FLG_START_LEN);
+      continue;
+    }
+
+    fixedUpCounter++;
+  }
+
+  // Zero up the fixed parameters that have been moved forward.
+  for (int i = fixedUpCounter; i < argc; i++) {
+    argv[i] = nullptr;
+  }
```

Comment:
Why aren't we using LLVM's command line processing? If you want a new command line flag, just register one with LLVM, there's no need to do custom argv processing.

Reply:
I did register a command and tried that, but the jit tool  that we delegate to still fails not recognizing the option. I had to take it out to make it work.